### PR TITLE
restruct the runcmdinstaller with netcat

### PR DESCRIPTION
**Pull request by immarvin**
replace awk with netcat;
refine the logic
refine the usage message

### CODE REVIEW

File: xCAT-server/sbin/runcmdinstaller [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+
+NSupdate=$(lsdef -t site -o clustersite -i nodestatus 2>/dev/null | grep "nodestatus="|cut -d= -f2) 
+nodestatus=$(lsdef $node -i status 2>/dev/null | grep "status="|cut -d= -f2)
 
-#print "sh " |& ns
-print action " " cmd " " |& ns
+if [[ "$NSupdate" != "n" && "$NSupdate" != "N" &&  "$NSupdate" != "0" && "$nodestatus" != "installing" ]];then
+    echo "Warning:"
+    echo "      The node status update is enabled and the 'nodestatus' of $node is $nodestatus"
+    echo "      runcmdinstaller can only be run against the node with 'installing' status!"
```
Comment (daniceexi) [inline xCAT-server/sbin/runcmdinstaller]: could the message be:
~~~~
runcmdinstaller can only be run against the node when it is in 'installing' status!
~~~~

File: xCAT-server/sbin/runcmdinstaller [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
+fi
 
-while((ns |& getline) > 0)
-print $0
+type -p nc >/dev/null 2>&1
+retcode=$?
+if [ "$retcode" != "0" ];then
+    echo "Error:"
+    echo "nc(netcat) not installed, please install it first!"
```
Comment (daniceexi) [inline xCAT-server/sbin/runcmdinstaller]: nc() is not installed.

**Comment by immarvin** [comment]
hi @daniceexi ,the code has been updated according to your comments

**Comment by daniceexi** [comment]
I have reviewed the change and agree to merge.

**Pull request merged**